torch/nn/modules/transformer.py

```diff
@@ -443,6 +448,11 @@ def forward(self, src: Tensor, src_mask: Optional[Tensor] = None,
         """
 
         # see Fig. 1 of https://arxiv.org/pdf/2002.04745v1.pdf
+        if src_key_padding_mask is not None:
+            if src_key_padding_mask.dtype != torch.bool:
+                if not torch.is_floating_point(src_key_padding_mask):
```

Comment:
Please consider dismantling pyramid of doom by a bit
```suggestion
        if src_key_padding_mask is not None:
            _skpm_dtype = src_key_padding_mask.dtype
            if _skpm_dtype != torch.bool and not _skpm_dtype.is_floating_point:
```